staging/src/k8s.io/client-go/util/workqueue/delaying_queue_test.go

```diff
@@ -154,6 +303,9 @@ func TestAddTwoFireEarly(t *testing.T) {
 	}
 
 	q.AddAfter(third, 2*time.Second)
+	if err := waitForWaitingQueueToFill(q); err != nil {
```

Comment:
is this a wait that was not previously necessary and now is? I hadn't dug into the implementation yet, but changes in behavior for existing callers not using the new functionality seems like an anti-goal to me

Reply:
Ah, no you are right, I was over-enthusiastically adding waitForWaitingQueueToFill calls after AddAfter with a delay and in this test there a waiting test that follows that would allow ample time for the channel to clear into the queu.e

Reply:
Have added it back as it was

Reply:
Ah right, no the test started failing - I had removed the processing of ready items out of the waitingLoop not seeing why it was there but now I see.  The TestAddTwoFireEarly test has the second item fire whilst adding new items which winds up with a negative nextReadyAt timer which will not fire (so we need the processing to catch the ready items next time around the loop).